test/unit/ResourceTypeRegistry.js

```diff
@@ -36,6 +54,24 @@ describe("ResourceTypeRegistry", function() {
     it("should be a getter/setter for a type",
```

Comment:
Can we get rid of this test? Because it's just not true any more that `type()` is a simple getter/setter, because we have all this default stuff now. (This test is only passing because `description` happens to match all the defaults.)